### PR TITLE
Unit test for jazz_metrics

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: core/jazz_metrics/test/test.js [hunk 1]
```diff
@@ -1,11 +1,1022 @@
...
+      index.genericValidation(event)
+        .catch(error => {
+          expect(error).to.include({
+            result: 'inputError',
+            message: 'Invalid Input Error'
+          })
+        });
+    });
+
+    it("should indicate method is invalid if empty or invalid metod is provided", () => {
```
Comment (suryajak) [inline core/jazz_metrics/test/test.js]: nitpick: method

File: core/jazz_metrics/index.js [hunk 2]
```diff
@@ -24,22 +24,13 @@ const errorHandlerModule = require("./components/error-handler.js"); //Import th
...
 const utils = require("./components/utils.js"); //Import the utils module.
 const validateUtils = require("./components/validation.js");
-const global_config = require("./config/global-config.json");
-
-module.exports.handler = (event, context, cb) => {
 
+function handler (event, context, cb) {
   var errorHandler = errorHandlerModule();
   var config = configObj(event);
-  var cloudwatch = new aws.CloudWatch({
```
Comment (suryajak) [inline core/jazz_metrics/index.js]: Looks like metrics is using older config reader, can you update it?

File: core/jazz_metrics/test/test.js [hunk 3]
```diff
@@ -1,11 +1,1022 @@
...
+          expect(error).to.include({
+            result: 'inputError',
+            message: 'Unsupported metric type.'
+          });
+          sinon.assert.calledOnce(getNameSpaceAndMetricDimensons);
+          getNameSpaceAndMetricDimensons.restore();
+        });
+    });
+
+    it("should indicate error if there is no assets available", () => {
```
Comment (suryajak) [inline core/jazz_metrics/test/test.js]: nitpick: "there are no assets"

File: core/jazz_metrics/test/test.js [hunk 4]
```diff
@@ -1,11 +1,1022 @@
...
+        invalidNameSpace.forEach(namespace => {
+          var resObj = utils.getNameSpaceAndMetricDimensons(namespace);
+          expect(resObj).to.have.all.keys('isError', 'awsNameSpace');
+          expect(resObj.isError).to.be.true;
+          expect(resObj.awsNameSpace).to.eq("Invalid");
+        })
+      })
+    });
+
+    it("should successfully get asset object for provided asset details", () => {
```
Comment (suryajak) [inline core/jazz_metrics/test/test.js]: How about validating for an asset type that is not supported yet? Send asset_type as `foo` and the API should return appropriate response (not supported)?

File: core/jazz_metrics/components/utils.js [hunk 5]
```diff
@@ -240,9 +240,26 @@ function updateCloudfrontAsset(newAssetObj, relativeId) {
...
+  var cloudwatch = new AWS.CloudWatch({
+    apiVersion: '2010-08-01'
+  });
+  return cloudwatch;
+}
+
+function cfCloudWatch() {
+  var cfcloudwatch = new AWS.CloudWatch({
+    apiVersion: '2010-08-01',
+    region: global_config.CF_REGION
```
Comment (suryajak) [inline core/jazz_metrics/components/utils.js]: I see that the value for CF_REGION is hard coded in global_config. It was a miss on the earlier review for metrics, so can you fix it?

File: core/jazz_metrics/components/utils.js [hunk 6]
```diff
@@ -240,9 +240,26 @@ function updateCloudfrontAsset(newAssetObj, relativeId) {
   return newAssetObj;
 }
 
+function initCloudWatch() {
```
[the remark this answers is not in the document]
Comment (supritat) [reply core/jazz_metrics/components/utils.js]: Replaced with `getCloudWatch()`

File: core/jazz_metrics/components/utils.js [hunk 7]
```diff
@@ -240,9 +240,26 @@ function updateCloudfrontAsset(newAssetObj, relativeId) {
...
+  var cloudwatch = new AWS.CloudWatch({
+    apiVersion: '2010-08-01'
+  });
+  return cloudwatch;
+}
+
+function cfCloudWatch() {
```
[the remark this answers is not in the document]
Comment (supritat) [reply core/jazz_metrics/components/utils.js]: Replaced with `getCloudfrontCloudWatch()`

File: core/jazz_metrics/test/test.js (hunk 1 above)
Comment (supritat) [reply core/jazz_metrics/test/test.js]: Updated

File: core/jazz_metrics/index.js (hunk 2 above)
Comment (supritat) [reply core/jazz_metrics/index.js]: Done

File: core/jazz_metrics/test/test.js (hunk 3 above)
Comment (supritat) [reply core/jazz_metrics/test/test.js]: Updated

File: core/jazz_metrics/test/test.js (hunk 4 above)
Comment (supritat) [reply core/jazz_metrics/test/test.js]: Included validation for asset type swagger_url and endpoint_url (Doesn't support for metrics).

File: core/jazz_metrics/components/utils.js (hunk 5 above)
Comment (supritat) [reply core/jazz_metrics/components/utils.js]: Since cloudFront region (CF_REGION) is always 'us-east-1', it is hard coded in global_config.

**Pull request merged**